src/prof.c

```diff
@@ -455,22 +455,25 @@ prof_sample_threshold_update(tsd_t *tsd) {
 	 * Compute sample interval as a geometrically distributed random
 	 * variable with mean (2^lg_prof_sample).
 	 *
-	 *                      __        __
-	 *                      |  log(u)  |                     1
-	 * bytes_until_sample = | -------- |, where p = ---------------
-	 *                      | log(1-p) |             lg_prof_sample
-	 *                                              2
+	 *                          |          |
+	 *                          |  log(u)  |                  1
```

Comment:
I think this is a little cleaner if we leave the math as is, and mention the zero->1 correction for the prng-output-is-zero case explicitly. Right now our logic seems sort of unmotivated if you only read the section from the book cited.

Reply:
Mention it in english down below I mean. E.g. "We actually apply a slight tweak to this; to handle the case in which the prng call outputs 0, we translate those values to be 1, effectively converting the distribution from [0, 1) to (0, 1]".

Reply:
Makes sense. Added explanations rather than tweaking the math, though I think the ideal way would be to explain in / change the book: the formula in the book is only correct if `u` is truly continuous, in which case `prob(u = 0) = prob(u = 1) = 0`; practitioners failing to pay attention to either one of the two ends would be in trouble.